### PR TITLE
Added new semester creation from the admin dashboard.

### PR DESCRIPTION
**Pull request by leahy-daniel**

### CODE REVIEW

File: app/controllers/admin/semester_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Admin::SemesterController < ApplicationController
+  def new
+  	@semester = Semester.new
```
Comment (chenhoward) [inline app/controllers/admin/semester_controller.rb]: We want to use 2 space indentations in our ruby code

File: app/helpers/admin/semester_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module Admin::SemesterHelper
```
Comment (chenhoward) [inline app/helpers/admin/semester_helper.rb]: Remove this file if it is not going to be used.

File: app/views/admin/semester/show.html.slim [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      h1
+        = @semester.year
+      h1
+        = @semester.lower_division_limit
+      h1
+        = @semester.upper_division_limit
+      h1
+        = @semester.application_deadline
+      h1
+        = @semester.acceptance_release_date
```
Comment (chenhoward) [inline app/views/admin/semester/show.html.slim]: New line at EOF

File: spec/controllers/admin/semester_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'spec_helper'
```
Comment (chenhoward) [inline spec/controllers/admin/semester_controller_spec.rb]: Remove this file if unused.

File: spec/helpers/admin/semester_helper_spec.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+require 'spec_helper'
```
Comment (chenhoward) [inline spec/helpers/admin/semester_helper_spec.rb]: Also remove this if it is unused as well

**Comment by chenhoward** [comment]
Also do validations with ability.rb (cancancan).  It will add an extra layer of security.

**Comment by leahy-daniel** [comment]
Removed the unwanted files and added some validation to the semesters.

**Comment by chenhoward** [comment]
@leahy-daniel ping me when you are done with the ability.rb stuff and want another review

File: app/models/user.rb [hunk 6]
```diff
@@ -124,7 +128,11 @@ def is_current_student?
     is_student_for_semester? Semester.current
   end
 
-
+  def is_instructor_for_semester(semester)
```
Comment (chenhoward) [inline app/models/user.rb]: With boolean returns we like to leave question marks on the method

File: app/views/admin/semester/show.html.slim [hunk 7]
```diff
@@ -0,0 +1,16 @@
+
+.container
+  .row
+    .col-md-12
+      h1
```
Comment (chenhoward) [inline app/views/admin/semester/show.html.slim]: Put labels on these

**Comment by leahy-daniel** [comment]
Changed the ability file to only include permissions to create a new semester for instructors, not for TAs.

**Comment by leahy-daniel** [comment]
@chenhoward Just finished the ability file, I think this is the way it was supposed to be done.

File: app/models/ability.rb [hunk 8]
```diff
@@ -11,8 +11,11 @@ def initialize(user)
     can :view, Lecture
     can :view, Assignment
```
Comment (chenhoward) [inline app/models/ability.rb]: Add a can :view, Semester because everyone should have the ability to see it, just not create update or destroy

**Comment by leahy-daniel** [comment]
Let everyone view the semester in the ability.rb file.

**Comment by chenhoward** [comment]
@leahy-daniel This LGTM.  Will be merged once a style is fixed.  Sorry I've been slow on getting to this I'm currently out of town and can't get internet consistently.

File: app/controllers/admin/semester_controller.rb [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    end
+  end
+
+  def show
+  end
+
+  private
+
+  def semester_params
+    params.require(:semester).permit(:semester, :year, :lower_division_limit, :upper_division_limit, :application_deadline, :acceptance_release_date)
```
Comment (chenhoward) [inline app/controllers/admin/semester_controller.rb]: Style: Break this to multiple lines.  We want to keep each line under 100 characters

```ruby
params.require(:semester).permit(
    :semester, :year, :lower_division_limit, :upper_division_limit,
    :application_deadline, :acceptance_release_date
)
```

**Comment by leahy-daniel** [comment]
@chenhoward No problem, thanks for the feedback.

**Comment by chenhoward** [comment]
@leahy-daniel Good work!  Merging now.  Will deploy soon.

**Comment by chenhoward** [comment]
Fix #165

**Pull request merged**